File: 251001/Palitykina/publisher/src/main/java/com/example/rest/service/MessageService/MessageService.java
package com.example.rest.service.MessageService;

import com.example.rest.dto.requestDto.MessageRequestTo;
import com.example.rest.dto.responseDto.MessageResponseTo;
import com.example.rest.dto.updateDto.MessageUpdateTo;

import java.util.List;

public interface MessageService{
    public List<MessageResponseTo> getAll();
    public MessageResponseTo get(long id);
    public MessageResponseTo create(MessageRequestTo input);
    public MessageResponseTo update(MessageUpdateTo input);
    public void delete(long id);

}

File: 251003/Zhuk/appLab2/src/main/kotlin/com/example/raw/dc/lab2/service/MessageService.kt
package com.example.raw.dc.lab2.service

import com.example.raw.dc.lab2.dto.request.MessageRequestTo
import com.example.raw.dc.lab2.dto.request.MessageRequestToId
import com.example.raw.dc.lab2.dto.response.MessageResponseTo

interface MessageService {
	suspend fun create(requestTo: MessageRequestTo?): MessageResponseTo?

	suspend fun deleteById(id: Long): Boolean

	suspend fun getAll(): List<MessageResponseTo>

	suspend fun getById(id: Long): MessageResponseTo?

	suspend fun update(requestTo: MessageRequestToId?): MessageResponseTo?
}